vkcs/compute/data_source_flavor.go

```diff
@@ -119,6 +127,98 @@ func DataSourceComputeFlavor() *schema.Resource {
 	}
 }
 
+type RequiredFlavor struct {
+	// Disk is the amount of root disk, measured in GB.
+	Disk    int  `json:"disk"`
+	HasDisk bool `json:"has_disk"`
+
+	// Disk is the amount of root disk, measured in GB.
+	MinDisk    int  `json:"min_disk"`
+	HasMinDisk bool `json:"has_min_disk"`
+
+	// RAM is the amount of memory, measured in MB.
+	RAM    int  `json:"ram"`
+	HasRAM bool `json:"has_ram"`
+
+	// MinRAM is the amount of memory, measured in MB.
+	MinRAM    int  `json:"min_ram"`
+	HasMinRAM bool `json:"has_min_ram"`
+
+	// Name is the name of the flavor.
+	Name    string `json:"name"`
+	HasName bool   `json:"has_name"`
+
+	// RxTxFactor describes bandwidth alterations of the flavor.
+	RxTxFactor    float64 `json:"rxtx_factor"`
+	HasRxTxFactor bool    `json:"has_rxtx_factor"`
+
+	// Swap is the amount of swap space, measured in MB.
+	Swap    int  `json:"-"`
+	HasSwap bool `json:"has_swap"`
+
+	// VCPUs indicates how many (virtual) CPUs are available for this flavor.
+	VCPUs    int  `json:"vcpus"`
+	HasVCPUs bool `json:"has_vcpus"`
+
+	// CPUGeneration is the cpu generation of flavor
+	CPUGeneration    string `json:"cpu_generation"`
+	HasCPUGeneration bool   `json:"has_cpu_generation"`
+
+	AccessType flavors.AccessType `json:"access_type"`
+}
+
+func NewRequiredFlavorFromResourceData(d *schema.ResourceData) *RequiredFlavor {
+	name, hasName := d.GetOk("name")
+	ram, hasRAM := d.GetOk("ram")
+	VCPUs, hasVCPUs := d.GetOk("vcpus")
+	disk, hasDisk := d.GetOk("disk")
+	minDisk, hasMinDisk := d.GetOk("min_disk")
+	minRAM, hasMinRAM := d.GetOk("min_ram")
+	rxTxFactor, hasRxTxFactor := d.GetOk("rx_tx_factor")
+	swap, hasSwap := d.GetOk("swap")
+	cpuGeneration, hasCPUGeneration := d.GetOk("cpu_generation")
+
+	if minRAM.(int) > ram.(int) {
+		ram = minRAM
+	}
+	if minDisk.(int) > disk.(int) {
+		disk = minDisk
+	}
+
+	accessType := flavors.AllAccess
+	if v, ok := d.GetOk("is_public"); ok {
+		if v, ok := v.(bool); ok {
+			if v {
+				accessType = flavors.PublicAccess
+			} else {
+				accessType = flavors.PrivateAccess
+			}
+		}
+	}
+
+	return &RequiredFlavor{
+		Disk:             disk.(int),
+		HasDisk:          hasDisk,
+		MinDisk:          minDisk.(int),
+		HasMinDisk:       hasMinDisk,
+		RAM:              ram.(int),
+		HasRAM:           hasRAM,
+		MinRAM:           minRAM.(int),
+		HasMinRAM:        hasMinRAM,
+		Name:             name.(string),
+		HasName:          hasName,
+		RxTxFactor:       rxTxFactor.(float64),
+		HasRxTxFactor:    hasRxTxFactor,
+		Swap:             swap.(int),
+		HasSwap:          hasSwap,
+		VCPUs:            VCPUs.(int),
+		HasVCPUs:         hasVCPUs,
+		CPUGeneration:    cpuGeneration.(string),
+		HasCPUGeneration: hasCPUGeneration,
+		AccessType:       accessType,
+	}
+}
+
```

Comment:
I believe it's a bit overkill to specify such a structure for filtering purposes, the older code is quite repetitive, and even if there are a few more lines in it, it's easy to read.

Reply:
But processing of all fields from schema is in one place,i have already write this code, there's no point in rolling it back

Reply:
Yup, you can leave the code as written but sometimes it's better to leave the code as is when there are no good reasons for changes.